### ISSUE TITLE
Improve user creation and membership flow

### ISSUE THREAD

**Issue by asmahood** [uwpokerclub/tickets#128]
#### Problem
Currently, there is a three step process in order to sign a first time member into an event.

1. User needs to be created on the users page
2. User needs to be added to the semester via the semesters page
3. Member needs to be signed into the event 

This requires an executive to visit three separate pages, and they have to refresh each page every time.

#### Solution
We should increase the functionality of the new membership form by allowing an executive member to either pick from a list of existing users, or to create a new user directly in that form. The flow would then be to create the user through the API first, and once that is finished, then create their membership through the API.

This should be the solution at the very minimum. A nice to have would be to have a toggle-able switch at the bottom of the form with some sort of label stating:
> Would you like to register this member into the latest event?

Toggling this to "Yes" would then register the new member into the most recently created event.
